lib/using_restricted_array.rb

```diff
@@ -1,61 +1,105 @@
-require_relative 'restricted_array.rb'
+require_relative "restricted_array.rb"
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: constant
 def length(array)
-  raise NotImplementedError
+  i = 0
+  while array[i] != nil
```

Comment:
Can `array` be `nil`? If so, will this work?

(Apply this comment to any methods you implemented in this PR.)